fsspec/dircache.py

```diff
@@ -1,9 +1,17 @@
+from __future__ import annotations
 import time
-from collections.abc import MutableMapping
+from typing import MutableMapping, List, TypedDict, Iterator, Any
+
 from functools import lru_cache
 
 
-class DirCache(MutableMapping):
+class DirEntry(TypedDict):
+    name: str
+    size: int
+    type: str
+
+
+class DirCache(MutableMapping[str, List[DirEntry]]):
```

Comment:
This class doesn't ever specify the key and value types, so theoretically I could make this a generic type. The docstrings imply this is the right type though?